main.rb

```diff
@@ -1,162 +1,29 @@
-    puts '5 - Create a rental'
-    puts '6 - List all rentals for a given person id'
-    puts '7 - Exit'
-  end
-
-  def list_books
-    @books.each { |book| puts book }
-  end
-
-  def list_people
-    @people.each { |person| puts person }
-  end
-
-  def create_person
-    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
-    choice = gets.chomp
-
-    case choice
-    when '1'
-      create_student
-    when '2'
-      create_teacher
-    else
-      puts 'That is not a valid input'
-      nil
-    end
-  end
-
-  def create_student
-    print 'Age: '
-    age = gets.chomp
-
-    print 'Name: '
-    name = gets.chomp
-
-    print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase == 'y'
-
-    student = Student.new(age: age, name: name, parent_permission: parent_permission, classroom: @classroom)
-    @people.push(student)
-
-    puts 'Person created successfully'
-  end
-
-  def create_teacher
-    print 'Age: '
-    age = gets.chomp
-
-    print 'Name: '
-    name = gets.chomp
-
-    print 'Specialization: '
-    specialization = gets.chomp
-
-    teacher = Teacher.new(age: age, name: name, specialization: specialization)
-    @people.push(teacher)
-
-    puts 'Person created successfully'
-  end
-
-  def create_book
-    print 'Title: '
-    title = gets.chomp
-
-    print 'Author: '
-    author = gets.chomp
-
-    book = Book.new(title, author)
-    @books.push(book)
-
-    puts 'Book created successfully'
-  end
-
-  def create_rental
-    puts 'Select a book from the following list by number'
-    @books.each_with_index { |book, i| puts "#{i}) #{book}" }
-    book_i = gets.chomp.to_i
-    puts
-    puts 'Select a person from the following list by number (not ID)'
-    @people.each_with_index { |person, i| puts "#{i}) #{person}" }
-    person_i = gets.chomp.to_i
-    puts
-    print 'Date: '
-    date = gets.chomp
-    rental = Rental.new(date, @books[book_i], @people[person_i])
-    @rentals.push(rental)
-    puts 'Rental created successfully'
-  end
-
-  def list_rentals
-    print 'ID of person: '
-    id = gets.chomp
-
-    puts 'Rentals:'
-    @rentals.each { |rental| puts rental if rental.person.id == id.to_i }
-  end
-end
-
-def main
-  app = App.new
-  app.run
-end
-
-main
+require_relative 'app'
+class Main
+  def initialize
+    @app = App.new
+  end
+  def actions
+    puts
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+  def run
+    puts 'Welcome to School Library App!'
+    loop do
+      actions
+      option = gets.chomp
+      break if option == '7'
+      @app.handle_action option
+    end
+    puts 'Thank you for using this app!'
+  end
+end
+run_application = Main.new
+run_application.run

```

Comment:
- [ ] The role of _main.rb_ is only to start their code and ask the user for the initial choice of options (or to quit). It should not have extra methods. You refactor your _main.rb_ in this manner:
```ruby
require "./app.rb"

def main
  app = App.new()

  response = nil

  puts "Welcome to School Library App!\n\n"
  while response != "7"
    puts "Please choose an option by enterin a number:"
    puts "1 - List all books"
    puts "2 - List all people"
    puts "3 - Create a person"
    puts "4 - Create a book"
    puts "5 - Create a rental"
    puts "6 - List all rentals for a given person id"
    puts "7 - Exit"
    response = gets.chomp

    case response
    when "1"
      app.list_books()
    when "2"
      app.list_people()
    when "3"
      app.create_person()
    when "4"
      app.create_book()
    when "5"
      app.create_rental()
    when "6"
      app.list_rentals_for_person_id()
    when "7"
      puts "Thank you for using this app!"
    end

    puts "\n"
  end
end

main()
```